rackcity/views/asset_views.py

```diff
@@ -166,18 +182,58 @@ def asset_add(request):  # need to make network and power connections here
             )
     if failure_message == "":
         try:
-            serializer.save()
-            return HttpResponse(status=HTTPStatus.CREATED)
+            asset = serializer.save()
         except Exception as error:
             failure_message += str(error)
-
+        else:
+            try:
+                save_mac_addresses(
+                    asset_data=data,
+                    asset_id=asset.id
+                )
+            except MacAddressException as error:
+                failure_message += "Some mac addresses couldn't be saved. " + \
+                    str(error)
```

Comment:
Only the invalid ones aren't saved, and the (user friendly) errors are listed here